### PR TITLE
fix(ui): Batch requests for stats on org dashboard (ISSUE-197)

### PR DESCRIPTION
**Pull request by billyvg**
The org dashboard attempts to load stats for all projects within a team. This PR batches the request so we only request up to 10 projects at a time.

### CODE REVIEW

File: src/sentry/static/sentry/app/actionCreators/projects.jsx [hunk 1]
```diff
@@ -60,20 +76,27 @@ const _debouncedLoadStats = debounce((api, projectSet, params) => {
...
-      ProjectActions.loadStatsForProjectSuccess(data);
-    },
-    error: data => {},
+  const queries = [
+    ...Array(Math.ceil(projects.length / MAX_PROJECTS_TO_FETCH)),
+  ].map((val, index) => {
+    const start = index * MAX_PROJECTS_TO_FETCH;
+    return _queryForStats(
+      api,
+      projects.slice(start, start + MAX_PROJECTS_TO_FETCH),
```
Comment (markstory) [inline src/sentry/static/sentry/app/actionCreators/projects.jsx]: Won't you fetch some projects twice with this?

* index = 0 gets 0-10
* index = 1 gets 10-20
* index = 2 gets 20 - 30

Should the max boundary be `start + MAX_PROJECTS_TO_FETCH - 1`?

File: src/sentry/static/sentry/app/actionCreators/projects.jsx [hunk 2]
```diff
@@ -60,20 +76,27 @@ const _debouncedLoadStats = debounce((api, projectSet, params) => {
...
+      params.orgId
+    );
   });
 
+  Promise.all(queries)
+    .then(results => {
+      ProjectActions.loadStatsForProjectSuccess(
+        results.reduce((acc, result) => [...acc, ...result], [])
```
Comment (markstory) [inline src/sentry/static/sentry/app/actionCreators/projects.jsx]: I think you ould use `acc.concat(result)` here :golf:

**Review by markstory — APPROVED** [review]

File: src/sentry/static/sentry/app/actionCreators/projects.jsx (hunk 1 above)
Comment (billyvg) [reply src/sentry/static/sentry/app/actionCreators/projects.jsx]: The end boundary doesn't get included `[0, 1, 2, 3].slice(0, 2)` will return `[0, 1]`

File: src/sentry/static/sentry/app/actionCreators/projects.jsx (hunk 2 above)
Comment (billyvg) [inline src/sentry/static/sentry/app/actionCreators/projects.jsx]:
```suggestion
        results.reduce((acc, result) => acc.concat(result), [])
```

File: src/sentry/static/sentry/app/actionCreators/projects.jsx (hunk 1 above)
Comment (markstory) [reply src/sentry/static/sentry/app/actionCreators/projects.jsx]: Ok. I missed the `.slice()` call my bad.

Comment (lynnagara) [reply src/sentry/static/sentry/app/actionCreators/projects.jsx]: chunk (from lodash) could simplify this code quite a bit

**Comment by jimgraham** [comment]
Thank you!

**Pull request merged**